src/lib.rs

```diff
@@ -833,6 +835,7 @@ where
         token_url: Option<TokenUrl>,
         userinfo_endpoint: Option<UserInfoUrl>,
         jwks: JsonWebKeySet<JS, JT, JU, K>,
+        id_token_signing_algs: Vec<JS>,
```

Comment:
I'd prefer not to introduce this as a breaking change to the public API. To avoid this, the `from_provider_metadata` function could either construct `Self` directly or this can be factored into a separate private `new_impl` method (either is fine).